dz2.2/dz2.2.ts

```diff
@@ -0,0 +1,389 @@
+
+    public set name(name: string) {
+        this._name = name;
+    }
+
+    public get surname(): string {
+        return this._surname;
+    }
+
+    public set surname(surname: string) {
+        this._surname = surname;
+    }
+
+    public get position(): string {
+        return this._position;
+    }
+
+    public set position(position: string) {
+        this._position = position;
+    }
+
+    public get company(): string {
+        return this._company;
+    }
+
+    public set company(company: string) {
+        this._company = company;
+    }
+
+    public get experience(): string {
+        return this._experience;
+    }
+
+    public set experience(experience: string) {
+        this._experience = experience;
+    }
+
+    public get courses(): string[] {
+        return this._courses;
+    }
+
+    public set courses(courses: string[]) {
+        this._courses = courses;
+    }
+
+    public get contacts(): string[] {
+        return this._contacts;
+    }
+
+    public set contacts(contacts: string[]) {
+        this._contacts = contacts;
+    }
+}
+
+class Area {
+    // implement getters for fields and 'add/remove level' methods
+    private _levels: Level[];
+    private _name: string;
+
+    constructor(levels: Level[], name: string) {
+        this._levels = levels;
+        this._name = name;
+    }
+
+    public get levels(): Level[] {
+        return this._levels;
+    }
+
+    public set levels(levels: Level[]) {
+        this._levels = levels;
+    }
+
+    public get name(): string {
+        return this._name;
+    }
+
+    public set name(name: string) {
+        this._name = name;
+    }
+
+    public addLevel(level: Level): number {
+        return this._levels.push(level);
+    }
+
+    public removeLevel(level: Level): number {
+        let newLevels: Level[] = [];
+        for (var i = 0; i < this._levels.length; i++) {
+            if (this._levels[i].name !== level.name) {
+                newLevels.push(this._levels[i]);
+            }
+        }
+
+        this._levels = newLevels;
+        return this._levels.length;
+    }
+}
+
+class Level {
+    // implement getters for fields and 'add/remove group' methods
+
+    private _groups: Group[];
+    private _name: string;
+    private _description: string;
+
+    constructor(groups: Group[], name: string, description: string) {
+        this._groups = groups;
+        this._name = name;
+        this._description = description;
+    }
+
+    public get groups(): Group[] {
+        return this._groups;
+    }
+
+    public set groups(groups: Group[]) {
+        this._groups = groups;
+    }
+
+    public get name(): string {
+        return this._name;
+    }
+
+    public set name(name: string) {
+        this._name = name;
+    }
+
+    public get description(): string {
+        return this._description;
+    }
+
+    public set description(description: string) {
+        this._description = description;
+    }
+
+    public addGroup(group: Group) {
```

Comment:
Please, do not forget to explicitly add type annotation for the return function value.

Reply:
Updated.